src/services/apiService.ts

```diff
@@ -0,0 +1,51 @@
+import { checkError, handleError } from "../utils/errorHelper";
+import localStoreService from "./localStoreService";
+
+const apiRoute = "https://reqres.in/api";
+
+export function anonymPostRequest(route: string, body: unknown) {
+  return request(route, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  });
+}
+
+export function authorizedPostRequest(route: string, body: unknown) {
+  return authorizedRequest(route, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  });
+}
+
+export function authorizedGetRequest(route: string) {
+  return authorizedRequest(route, {});
+}
+
+export function anonymGetRequest(route: string) {
+  return request(route, {});
+}
+
+function request(route: string, config: RequestInit) {
+  return fetch(`${apiRoute}${route}`, config)
+    .then(checkError)
+    .then((response) => response.json())
+    .catch(handleError);
+}
+
+function authorizedRequest(route: string, config: RequestInit) {
+  const authToken = localStoreService.getAuthToken();
+  if (!authToken) {
+    throw console.warn("no token");
+  }
+  config.headers = {
+    ...config?.headers,
+    Authorization: `Bearer ${localStoreService.getAuthToken()}`,
```

Comment:
You already work with `access token` here. Do you think you will have to cope with `refresh token` in addition? Is there any need to worry about that in this particular PR?

Reply:
as I understand `refresh token` must be generated on the backend and can be stored in cookies with HTTPS only flag, so I may not need to implement this part.

Reply:
Not what I meant - in case auth flow expects presence of refresh token you need to think of away to rerequest access token when expired (usually based on 401 server response)

Reply:
mocked api doesn’t provide opportunities to work with a refresh token, so for now i will not implement this logic, but `authorizeDecorator` can be extended in future